File: PO_Projekt3/owca.py
from zwierze import Zwierze

class Owca(Zwierze):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 4, 4, 1.00, 1, 0.00, 1.00)

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Owca)

    def stworzDziecko(self, cel):
        return Owca(self._swiat, cel)

    def rysowanie(self):
        return (153, 230, 255)

File: projekty/PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Zwierzeta/Owca.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Zwierzeta;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Owca extends Zwierze {
	public Owca(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Owca.class), Stale.inicjatywa.get(Owca.class),
				Stale.szansaRuchu.get(Owca.class), Stale.zasiegRuchu.get(Owca.class),
				Stale.szansaUcieczki.get(Owca.class), Stale.szansaRozmnazania.get(Owca.class));
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Owca;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Owca(swiat, cel);
	}
}
